src/communication/bolt/v1/decoder/decoder.hpp

```diff
@@ -131,6 +131,18 @@ class Decoder {
             return ReadUnboundedEdge(data);
           case Signature::Path:
             return ReadPath(data);
+          case Signature::DateTime:
+            return ReadDateTime(data);
+          case Signature::DateTimeZoneId:
+            return ReadDateTimeZoneId(data);
+          case Signature::LegacyDateTime: {
+            if (major_v_ > 4) return false;
```

Comment:
1. Per the [documentation](https://memgraph.com/docs/client-libraries) Memgraph does provide backward compatibility. In any case, we should support the pre-Bolt v5 structures because mgclient is v4.
2. The latest versions of mgconsole/Lab still open Bolt v4 connections, which means that Bolt uses the legacy structures for their values.
3. The legacy vs. latest distinction only applies to the Bolt protocol. In Memgraph, they are represented by the same type.
4. No (legacy datetime is a structure for zoned temporal values, and my PR is the one that will add support for them)